moped-editor/src/utils/mapDrawHelpers.js

```diff
@@ -212,24 +211,82 @@ const findDifferenceByFeatureProperty = (featureProperty, arrayOne, arrayTwo) =>
  * @property {function} setIsDrawing - Toggle draw tools
  * @property {function} renderMapDrawTools - Function that returns JSX for the draw tools in the map
  * @property {function} saveDrawnPoints - Function that saves features drawn in the UI
+ * @property {function} saveActionDispatch - Function that helps us send signals to other components
  */
 export function useMapDrawTools(
   featureCollection,
   setFeatureCollection,
   projectId,
   refetchProjectDetails,
-  currentZoom
+  currentZoom,
+  saveActionDispatch
 ) {
-  const isNewProject = projectId === null;
-
   const mapEditorRef = useRef();
   const [isDrawing, setIsDrawing] = useState(false);
   const [modeId, setModeId] = useState(null);
   const [modeHandler, setModeHandler] = useState(null);
-  const [selectedFeatureIndex, setSelectedFeatureIndex] = useState(null);
-  const [selectedEditHandleIndexes, setSelectedEditHandleIndexes] = useState(
-    []
-  );
+
+  /**
+   * Returns true if lineStringA has the same coordinates as lineStringB
+   * @param {Object} lineStringA - A line string feature object to compare
+   * @param {Object} lineStringB - Another line string feature object to compare against
+   * @return {boolean}
+   */
+  const lineStringEqual = (lineStringA, lineStringB) => {
+    const lineStringCoordinatesA = lineStringA?.geometry?.coordinates ?? [];
+    const lineStringCoordinatesB = lineStringB?.geometry?.coordinates ?? [];
+
+    return (
+      // Test both have more than one pair of coordinates
+      lineStringCoordinatesA.length > 1 &&
+      lineStringCoordinatesB.length > 1 &&
+      // Test they both have the same length
+      lineStringCoordinatesA.length === lineStringCoordinatesB.length &&
+      // Test every element in both lines is equivalent
+      lineStringCoordinatesA.every(
+        (coordinatePair, currentIndex) =>
+          coordinatePair[0] === lineStringCoordinatesB[currentIndex][0] &&
+          coordinatePair[1] === lineStringCoordinatesB[currentIndex][1]
+      ) // 'every' returns true if every comparison (the predicate) in the array returns true
+    );
+  };
+
+  /**
+   * Returns true if lineStringA has the same coordinates as lineStringB
+   * @param featurePointA - A point to compare
+   * @param featurePointB - Another point to compare against
+   * @return {boolean}
+   */
+  const pointEqual = (featurePointA, featurePointB) => {
```

Comment:
for future refactoring: turfjs my help here: https://turfjs.org/docs/#booleanEqual